lib/simple_token_authentication.rb

```diff
@@ -46,7 +46,7 @@ def self.load_available_adapters adapters_short_names
   end
 
   def self.adapter_dependency_fulfilled? adapter_short_name
-    qualified_const_defined?(SimpleTokenAuthentication.adapters_dependencies[adapter_short_name])
+    const_defined?(SimpleTokenAuthentication.adapters_dependencies[adapter_short_name])
```

Comment:
Hi @gabrielc63,

I'm figuring out that this is the only change that breaks backward-compatibility when adding support for Rails 5.

This method call is used to check if `ActiveRecord::Base`, for example, is defined before loading the corresponding adapter. It happens that `Module#const_defined?` as defined in Ruby 1.9.3 doesn't consider `ActiveRecord::Base` to be a valid constant name (`ActiveRecord` and `Base` are valid by themselves). That's why Rails did define `Module#qualified_const_defined?`, which considers `ActiveRecord::Base` as a valid _qualified_ constant name and [recursively delegates][src] to `const_defined?`.

Ruby 2 extended its definition of `const_defined?` to add support for _qualified_ constant names. Since Rails 5 drops support for Ruby 1.9.3, there is no longer use for `qualified_const_defined?`, hence the deprecation.

My first idea was to drop support for Ruby 1.9.3 as well. A non-backward compatible change which [would have required][semver] to release _Simple Token Authentication_ **v2.0.0**.

However, scoping this change to Rails 5 users would allow to support Rails 5 without dropping support for Rails 4 / Ruby 1.9.3, which is a much better option in exchange of minimal added complexity. So, I won't include this change as [I initially intended to][intent], but I'm grateful to you for this PR which pointed me in the right direction. Again, thank you!

  [intent]: https://github.com/gonzalo-bulnes/simple_token_authentication/issues/214#issuecomment-195699806
  [semver]: http://semver.org/
  [src]: https://github.com/rails/rails/blob/432a65fab2a6c7eb6ff77062e73f7627470f7da7/activesupport/lib/active_support/core_ext/module/qualified_const.rb#L26-L33